src/rapier_body_utils_2d.cpp

```diff
@@ -110,12 +103,10 @@ bool RapierBodyUtils2D::body_motion_recover(
 				}
 			}
 		}
-
 		if (recovered) {
 			p_recover_motion += recover_step;
 			p_transform.columns[2] += recover_step;
 			p_body_aabb.position += recover_step;
```

Comment:
Again, I'm just looking at the changes, pretty hard to do, but the principle is to move the body, except here you're moving the margin.

see at the end (the body position here is not moved by the recover):

```
		// Get the rest info in unsafe advance
		Vector2 unsafe_motion = p_motion * best_unsafe;
		body_transform.columns[2] += unsafe_motion;
		body_aabb.position += unsafe_motion;
```